### ISSUE TITLE
[enhancement]: Allow `NoCloud` to specify alternative `cidata` fs label.

### ISSUE THREAD

**Issue by mjsir911** [canonical/cloud-init#5532]
# Enhancement

This is kind of a regression since I've been depending on behaviour that was recently removed.

Basically I want the behaviour of `fs_label` back. When configuring custom raspberry pi images to be bootable with `cloud-init`, it's useful to just drop the cloud config files into `/boot/` (with vfat label `boot`) as opposed to having to provision a whole new partition solely for cloud-init.

Related: 7a686bfac9f17745c40831b01f54438a4d78f161 (#5515) #386 #3694 #5062

**Comment by holmanb** [comment]
@mjsir911 please let me know if your use case isn't addressed by one of the suggested workarounds

**Comment by mjsir911** [comment]
Thank you all for the prompt response! `seedfrom: file://boot/` __should__ do what I want, I'll confirm this but I appreciate the plain outlining of alternatives :-)

**Comment by holmanb** [comment]
> Thank you all for the prompt response! `seedfrom: file://boot/` **should** do what I want, I'll confirm this but I appreciate the plain outlining of alternatives :-)

Of course! Happy to help. Let me know how it goes.

**Comment by holmanb** [comment]
Happy to help! Let me know if that doesn't solve it for you :-)

@mjsir911 Also quick question: do you already have something like `datasource_list: [NoCloud]` set in your base configuration? If not, you might be able to optimize something.

If you don't already set `datasource_list` in your config, then cloud-init is likely taking unnecessary time during boot attempting to find a configuration from datasources besides NoCloud. Once [23be88d0](https://github.com/canonical/cloud-init/commit/23be88d005983aa39b248ad55b1a3008973673c6#diff-8e27dfc14b3ffceee1e29d3cf58dc25309c0ac070879febd47c578de04ea2ddd) gets released (likely 24.3), this should be automatically fixed if you use `seedfrom: file://boot/`. The way to check if cloud-init is wasting time is to check for `DataSourceMAAS` and `DataSourceConfigDrive` in your `/etc/cloud/cloud-init.log` - if it doesn't exist then you won't benefit from this optimization.

**Comment by mjsir911** [comment]
> @mjsir911 Also quick question: do you already have something like datasource_list: [NoCloud] set in your base configuration? If not, you might be able to optimize something.

Yeah, copying from [another pi-cloud-init project](https://github.com/timebertt/pi-cloud-init), I have:
```
datasource_list: [ NoCloud, None ]
datasource:
  NoCloud:
...
```

which seems to not be hanging but I'm still tinkering with loading up `user-data.yaml` files & such

**Comment by holmanb** [comment]
The issue raised has been addressed.
